File: fizz_buzz.nim
import strformat

for i in countup(1, 100):
    let div3 = (i mod 3 == 0)
    let div5 = (i mod 5 == 0)
    echo(if div3: (if div5: "FizzBuzz" else: "Fizz") else: (if div5: "Buzz" else: fmt"{i}"))

File: fizz_buzz.zig
//
// fizz_buzz.zig
// Based on https://ziglang.org/learn/samples/#zigg-zagg
//
// Distributed under terms of the MIT license.
//

const std = @import("std");

pub fn main() !void {
    const stdout = std.io.getStdOut().writer();
    var i: usize = 1;
    while (i <= 100) : (i += 1) {
        if (i % 15 == 0) {
            try stdout.writeAll("FizzBuzz\n");
        } else if (i % 3 == 0) {
            try stdout.writeAll("Fizz\n");
        } else if (i % 5 == 0) {
            try stdout.writeAll("Buzz\n");
        } else {
            try stdout.print("{d}\n", .{i});
        }
    }
}
